nncf/config/extractors.py

```diff
@@ -44,6 +44,10 @@ def extract_algo_specific_config(config: NNCFConfig, algo_name_to_match: str) ->
     :return: The sub-dictionary, exactly as it is specified in the NNCF configuration of the .json file,
     that corresponds to the algorithm-specific data (i.e. {"algorithm": "quantization", ... })
     """
+    from nncf.common.compression import NO_COMPRESSION_ALGORITHM_NAME
+    if algo_name_to_match == NO_COMPRESSION_ALGORITHM_NAME:
```

Comment:
It looks like a workaround. What do you think about creating a method to validate config and call it in each algorithm, because not all algorithms require this check.